src/sst/core/CMakeLists.txt

```diff
@@ -0,0 +1,244 @@
+# retains certain rights in this software.
+#
+# See LICENSE for full license details
+#
+
+add_definitions(-DSST_BUILDING_CORE=1)
+
+# -- Required for build
+add_subdirectory(impl)
+add_subdirectory(model)
+add_subdirectory(shared)
+add_subdirectory(sync)
+add_subdirectory(testingframework)
+add_subdirectory(testElements)
+
+# -- Required for header installation
+add_subdirectory(cfgoutput)
+add_subdirectory(eli)
+add_subdirectory(env)
+add_subdirectory(interfaces)
+add_subdirectory(interprocess)
+add_subdirectory(math)
+add_subdirectory(part)
+add_subdirectory(rng)
+add_subdirectory(serialization)
+add_subdirectory(statapi)
+
+add_library(
+  sst-core-lib OBJECT
+  action.cc
+  clock.cc
+  baseComponent.cc
+  component.cc
+  componentExtension.cc
+  componentInfo.cc
+  config.cc
+  configGraph.cc
+  cfgoutput/pythonConfigOutput.cc
+  cfgoutput/dotConfigOutput.cc
+  cfgoutput/xmlConfigOutput.cc
+  cfgoutput/jsonConfigOutput.cc
+  eli/elibase.cc
+  eli/elementinfo.cc
+  elemLoader.cc
+  event.cc
+  exit.cc
+  factory.cc
+  heartbeat.cc
+  initQueue.cc
+  link.cc
+  memuse.cc
+  oneshot.cc
+  output.cc
+  params.cc
+  pollingLinkQueue.cc
+  simulation.cc
+  subcomponent.cc
+  sharedRegion.cc
+  timeLord.cc
+  uninitializedQueue.cc
+  unitAlgebra.cc
+  module.cc
+  sstpart.cc
+  timeVortex.cc
+  serialization/serializable.cc
+  serialization/serialize_serializable.cc
+  serialization/serializer.cc
+  serialization/statics.cc
+  interfaces/TestEvent.cc
+  interfaces/simpleMem.cc
+  interfaces/stdMem.cc
+  interfaces/simpleNetwork.cc
+  interprocess/ipctunnel.cc
+  interprocess/mmapparent.cc
+  interprocess/shmparent.cc
+  rng/marsaglia.cc
+  rng/mersenne.cc
+  rng/xorshift.cc
+  statapi/statengine.cc
+  statapi/statgroup.cc
+  statapi/statoutput.cc
+  statapi/statfieldinfo.cc
+  statapi/statoutputconsole.cc
+  statapi/statoutputtxt.cc
+  statapi/statoutputcsv.cc
+  statapi/statoutputjson.cc
+  statapi/statbase.cc
+  cputimer.cc
+  iouse.cc)
+
+set(SSTHeaders
+    action.h
+    activity.h
+    activityQueue.h
+    baseComponent.h
+    clock.h
+    componentExtension.h
+    component.h
+    componentInfo.h
+    configGraph.h
+    configGraphOutput.h
+    config.h
+    cputimer.h
+    decimal_fixedpoint.h
+    elemLoader.h
+    event.h
+    exit.h
+    factory.h
+    from_string.h
+    heartbeat.h
+    initQueue.h
+    iouse.h
+    link.h
+    linkMap.h
+    linkPair.h
+    mempool.h
+    memuse.h
+    module.h
+    objectComms.h
+    oneshot.h
+    output.h
+    params.h
+    pollingLinkQueue.h
+    profile.h
+    rankInfo.h
+    sharedRegion.h
+    sharedRegionImpl.h
+    simulation.h
+    sparseVectorMap.h
+    sstinfo.h
+    sstpart.h
+    sst_types.h
+    stopAction.h
+    stringize.h
+    subcomponent.h
+    threadsafe.h
+    timeConverter.h
+    timeLord.h
+    timeVortex.h
+    uninitializedQueue.h
+    unitAlgebra.h
+    warnmacros.h)
+
+target_include_directories(sst-core-lib PUBLIC ${sst-core_SOURCE_DIR}/../src)
```

Comment:
These files are installed. You need to use the BUILD_INTERFACE/INSTALL_INTERFACE generator expressions for target_link_libraries to work correctly in downstream projects.

Reply:
Will do in future PR.